### PR TITLE
Return corrupt in the event a response cache entry can't be read

### PR DESCRIPTION
**Pull request by spencerkimball**

### CODE REVIEW

**Comment by tamird** [comment]
Can you add a test?

File: storage/range.go [hunk 1]
```diff
@@ -880,8 +880,8 @@ func (r *Range) applyRaftCommandInBatch(ctx context.Context, index uint64, origi
 			// We successfully read from the response cache, so return whatever error
 			// was present in the cached entry (if any).
 			return batch, reply.Header().GoError()
-		} else if ok && err != nil {
-			// If the response cache entry was found but couldn't be decoded, return corrupt.
+		} else if err != nil {
```
Comment (bdarnell) [inline storage/range.go]: I'd swap the two clauses to put `if err !=nil` first and then `if ok`

**Comment by bdarnell** [comment]
LGTM

File: storage/range.go (hunk 1 above)
Comment (spencerkimball) [inline storage/range.go]: Done.

**Pull request merged**